File: miscellaneous/graph-bfs/20221219/graph_bfs.py
from typing import List


def graph_bfs(graph: List[List[int]]) -> List[int]:
    queue = [0]
    values = []

    while len(queue):
        vertex = queue.pop(0)
        values.append(vertex)

        for neighbor in graph[vertex]:
            if neighbor not in values:
                queue.append(neighbor)

    return list(values)

File: miscellaneous/graph-bfs/20221219/graph-bfs.php
<?php

declare(strict_types=1);

function graphBfs(array &$graph): array
{
    $queue = [0];
    $values = [];

    while (count($queue)) {
        $vertex = array_shift($queue);
        $values[] = $vertex;

        foreach($graph[$vertex] as $neighbor) {
            if (!in_array($neighbor, $values)) {
                $queue[] = $neighbor;
            }
        }
    }

    return $values;
}